nanoFramework.WebServer/WebServer.cs

```diff
@@ -537,35 +534,69 @@ private void StartListener()
                         // Variables used only within the "for". They are here for performance reasons
                         bool mustAuthenticate;
                         bool isAuthOk;
-                        bool isRoute = false;
+                        CallbackRoutes selectedRoute = null;
+                        string multipleCallback = null;
 
                         foreach (CallbackRoutes route in _callbackRoutes)
                         {
                             if (!IsRouteMatch(route, context.Request.HttpMethod, context.Request.RawUrl))
                             {
                                 continue;
                             }
+                            if (selectedRoute is null)
+                            {
+                                selectedRoute = route;
+                            }
+                            else
+                            {
+                                multipleCallback ??= $"Multiple matching callbacks: {selectedRoute.Callback.DeclaringType.FullName}.{selectedRoute.Callback.Name}";
+                                multipleCallback += $", {route.Callback.DeclaringType.FullName}.{route.Callback.Name}";
+                            }
+                        }
 
-                            isRoute = true;
+                        if (multipleCallback is not null)
```

Comment:
Can this be done on DEBUG only? a bit like in the full .NET where errors appears only in debug mode?

Reply:
Well, at the same time, it is compiled in Released mode in the nuget

Reply:
Yes  that's the reason. The message for the internal server error is the only communication channel available. AFAIK there is no way to control the output of the class library depending on the configuration of the main program in .NET, so I guess that's the same for nanoFramework. In .NET there's trace as alternative for debug messages, but not in nanoframework.

Reply:
ok, fine with that then. I don't want to add another element in the lib. Let's then display this error message when multiple functions with the same route are found.